UnityWeld/Binding/Internal/TypeResolver.cs

```diff
@@ -310,5 +366,73 @@ public static BindableMember<PropertyInfo>[] FindBindableCollectionProperties(Co
                 .Where(p => !typeof(string).IsAssignableFrom(p.Member.PropertyType))
                 .ToArray();
         }
+
+        /// <summary>
+        /// Register custom binding attribute.
+        /// This will allow to mark bindable properties in external dlls without referencing UnityWeld.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        public static void RegisterCustomBindingAttribute<T>() where T : Attribute
+        {
+            var type = typeof(T);
+            if (BindingAttributeTypes.Contains(type))
+            {
+                return;
+            }
+
+            BindingAttributeTypes.Add(type);
+        }
+
+        /// <summary>
+        /// Register custom ViewModel provider.
+        /// This will allow to use custom ViewModel providers in external dlls without referencing UnityWeld.
+        /// </summary>
+        /// <param name="provider"></param>
+        public static void RegisterCustomViewModelProvider(ViewModelProviderDelegate provider)
+        {
+            if (ViewModelProviders.Contains(provider))
+            {
+                return;
+            }
+
+            ViewModelProviders.Add(provider);
+        }
+
+        /// <summary>
+        /// Check if type has binding attribute
+        /// </summary>
+        public static bool HasBindingAttribute(this MemberInfo type)
+        {
+            //for to avoid GC
+            for (var index = 0; index < BindingAttributeTypes.Count; index++)
+            {
+                if (type.GetCustomAttributes(BindingAttributeTypes[index], false).Any())
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Get ViewModel data from component
+        /// </summary>
+        public static ViewModelProviderData GetViewModelData(this Component component)
+        {
+            if (component == null)
+                return null;

```

Comment:
code style - we always use brackets around single-line `if`, `for` and `while` statements